File: dataset/output/OPENAI_gpt-4o/Task33_OPENAI_gpt-4o.cpp
#include <iostream>
#include <queue>
#include <vector>

int queueTime(std::vector<int> customers, int n) {
    std::priority_queue<int, std::vector<int>, std::greater<int>> tills;
    for (int i = 0; i < n; i++) {
        tills.push(0);
    }
    for (int time : customers) {
        int earliestTill = tills.top();
        tills.pop();
        earliestTill += time;
        tills.push(earliestTill);
    }
    int totalTime = 0;
    while (!tills.empty()) {
        totalTime = tills.top();
        tills.pop();
    }
    return totalTime;
}

int main() {
    std::cout << queueTime({5, 3, 4}, 1) << std::endl; // should return 12
    std::cout << queueTime({10, 2, 3, 3}, 2) << std::endl; // should return 10
    std::cout << queueTime({2, 3, 10}, 2) << std::endl; // should return 12
    return 0;
}

File: dataset/output/OPENAI_gpt-4o/task33/Task33_OPENAI_gpt_4o.java
import java.util.PriorityQueue;

public class Task33_OPENAI_gpt_4o {
    public static int queueTime(int[] customers, int n) {
        PriorityQueue<Integer> tills = new PriorityQueue<>();
        for (int i = 0; i < n; i++) {
            tills.add(0);
        }
        for (int time : customers) {
            int earliestTill = tills.poll();
            earliestTill += time;
            tills.add(earliestTill);
        }
        int totalTime = 0;
        while (!tills.isEmpty()) {
            totalTime = tills.poll();
        }
        return totalTime;
    }

    public static void main(String[] args) {
        System.out.println(queueTime(new int[]{5, 3, 4}, 1)); // should return 12
        System.out.println(queueTime(new int[]{10, 2, 3, 3}, 2)); // should return 10
        System.out.println(queueTime(new int[]{2, 3, 10}, 2)); // should return 12
    }
}